org.eclipse.winery.frontends/app/tosca-management/src/app/instance/sharedComponents/interfaces/interfaces.component.ts

```diff
@@ -342,6 +354,57 @@ export class InterfacesComponent implements OnInit {
             );
     }
 
+    isLoading = () => Utils.isLoading(this._loading) || this.loading;
+
+    toggleDiv(par: InheritedInterface) {
+        par.is_shown = !par.is_shown;
+    }
+
+    process_url(parentType: string) {
+        const prc = parentType.replace('{', '').split('}');
+        prc[0] = Utils.nodeTypeURL(parentType);
+        return prc;
+
+    }
+
+    override_interface(inh: InterfacesApiData) {
+        this.interfacesData.push(inh);
+    }
+
+    interface_exist(inh: InterfacesApiData) {
+        const filtered_interface: InterfacesApiData[] = this.interfacesData.filter((value) => value.name === inh.name);
+        return filtered_interface.length === 0;
+    }
+
+    override_operation(inh: InterfacesApiData, op: InterfaceOperationApiData) {
+        const filtered_interface: InterfacesApiData[] = this.interfacesData.filter((value) => value.name === inh.name);
```

Comment:
Instead of filter, you can use find. Then get a single interface and not an array.